### PR TITLE
fix(amplify_authenticator): fix country code bugs

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:* https://github.com/aws-amplify/amplify-flutter/issues/1296

*Description of changes:*
- Store country code state in AuthenticatorState (formally view model) to persist it between sign up and confirm sign up
- Update phone number in AuthenticatorState when the country code changes
- Add tests for non US numbers
- Clean up tests (remove legacy tests, rename a test file)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart [hunk 1]
```diff
@@ -158,9 +157,7 @@ mixin AuthenticatorPhoneFieldMixin<FieldType,
     );
 
     if (selectedCountry != null) {
-      setState(() {
-        _selectedCountry = selectedCountry;
-      });
+      state.country = selectedCountry;
```
Comment (dnys1) [inline packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart:160]: Seems like `setState` would still be needed, no?

File: packages/amplify_authenticator/example/integration_test/utils/mock_data.dart [hunk 2]
```diff
@@ -34,10 +34,54 @@ String generatePassword() =>
...
-String randomPhoneDigits() => '${random.nextInt(10)}${random.nextInt(10)}';
+String randomPhoneDigits(int count) {
+  String value = '';
+  for (var i = 0; i < count; i++) {
+    value += random.nextInt(10).toString();
+  }
+  return value;
+}
+
+class _PhoneNumber {
```
Comment (dnys1) [inline packages/amplify_authenticator/example/integration_test/utils/mock_data.dart:67]: Seems like this might be a useful class in Authenticator, instead of working with strings

**Review by dnys1 — APPROVED** [review]
Looks good, few minor Qs

**Review by haverchuck — APPROVED** [review]

File: packages/amplify_authenticator/example/integration_test/utils/mock_data.dart (hunk 2 above)
Comment (Jordan-Nelson) [reply packages/amplify_authenticator/example/integration_test/utils/mock_data.dart:67]: Agreed. I am a little hesitant to do this right now though. I started to, but it got messy quickly. I think we need to do some (not all) of the refactoring work around AuthenticatorFormField I proposed in https://github.com/aws-amplify/amplify-flutter/pull/1116. This would make the form field code easier to understand. In particular, the current definition of AuthenticatorFormField I think can be simplified.

The current definition is: 

```dart
abstract class AuthenticatorFormField<FieldType, FieldValue,
        T extends AuthenticatorFormField<FieldType, FieldValue, T>>
    extends AuthenticatorComponent<T> {
}
```

IMO it is hard to follow what FieldType, FieldValue, and T are in this context. I am pretty sure we could simplify it to something like:

```dart
abstract class AuthenticatorFormField<FieldType>
    extends AuthenticatorComponent {
}
```

In the above definition, FieldType would be the data type for the data in the form field (String, PhoneNumber, etc.). In the previous refactor PR I put up, I simplified it further remove both `FieldType` and `FieldValue`. While that worked, it meant that every field was a String, which isn't what we want.

File: packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart:160]: The country is now stored in AuthenticatorState and the setter for country calls `notifyListeners()`.
